### DISCUSSION TITLE
How to group javascript lines?

### DISCUSSION THREAD

**Discussion by pvanthony** [romcal/romcal#312] [Q&A] [answered]
This a pure javascript question. Only answer if there is time.

There is this bunch of code that repeats often. Hoping to only have one place to make changes in the future.
```
        if (result[day].rank == "SOLEMNITY" || result[day].rank == "FEAST"){
            var calenderLink = url+ result[day].rank + "/" + result[day].cycles.sundayCycle + "/" + result[day].key
            mydescription += "<a href=\'" + calenderLink + ".mp3\' target=\'_blank\'><span style=\"font-size:12px;\">ஒலி</span></a><br>";
            mydescription += "<a href=\'" + calenderLink + ".mp4\' target=\'_blank\'><span style=\"font-size:12px;\">ஒளி</span></a>&nbsp;&nbsp;&nbsp;&nbsp;";
            mydescription += "<a href=\'" + calenderLink + "R.mp3\' target=\'_blank\'><span style=\"font-size:12px;\">சிந்தனை</span><br></a>"
            if (result[day].calendar.dayOfWeek == 0){
                mydescription += "<a href=\'" + calenderLink + "M.pdf\' target=\'_blank\'><span style=\"font-size:12px;\">திருப்பலி</span><br></a>"
              }
        }
        else if (result[day].rank == "SUNDAY"){
            var calenderLink = url+ result[day].seasons + "/" + "SUNDAY" + "/" + result[day].cycles.sundayCycle + "/" + String(result[day].calendar.weekOfSeason).padStart(2, '0') + "/" + result[day].calendar.dayOfWeek
            mydescription += "<a href=\'" + calenderLink + ".mp3\' target=\'_blank\'><span style=\"font-size:12px;\">ஒலி</span></a><br>";
            mydescription += "<a href=\'" + calenderLink + ".mp4\' target=\'_blank\'><span style=\"font-size:12px;\">ஒளி</span></a>&nbsp;&nbsp;&nbsp;";
            mydescription += "<a href=\'" + calenderLink + "R.mp3\' target=\'_blank\'><span style=\"font-size:12px;\">சிந்தனை</span><br></a>";
            mydescription += "<a href=\'" + calenderLink + "M.pdf\' target=\'_blank\'><span style=\"font-size:12px;\">திருப்பலி</span><br></a>";
        }
```
This his the repeating line.
```
            mydescription += "<a href=\'" + calenderLink + ".mp3\' target=\'_blank\'><span style=\"font-size:12px;\">ஒலி</span></a><br>";
            mydescription += "<a href=\'" + calenderLink + ".mp4\' target=\'_blank\'><span style=\"font-size:12px;\">ஒளி</span></a>&nbsp;&nbsp;&nbsp;&nbsp;";
            mydescription += "<a href=\'" + calenderLink + "R.mp3\' target=\'_blank\'><span style=\"font-size:12px;\">சிந்தனை</span><br></a>"
```
Is there something in javascript that can take this three lines and put them in a variable that can be used in other place in the code? Please note that the three lines has a variable called "calenderLink".

**Comment by emagnier** [comment, marked as answer]
Hello @pvanthony,

For this kind of questions, the better place is https://stackoverflow.com/.
This GitHub repository / discussions must stay focused only on romcal.

Regarding your question, you will probably find what you are looking for here in [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals). 
Or just do a Google search for "javascript string template literal".
Thank you for your understanding 😉

**Reply by pvanthony** [reply]
You are correct. My bad for sending the question here. In the future my questions will be related to romcal.
I completely understand this. My bad.

Thank you for the link to the article. I think it is exactly what I needed. I will try it out. Thank you again for helping.

**Reply by emagnier** [reply]
No worries, I wish you good continuation in your project.

**Reply by tukusejssirs** [reply]
Although @emagnier is right that this is not the right place for questions not related with romcal, but here is another suggestion to fix the current problem of code duplication: use `if` conditions (or use a `switch` if there are multiple possible values of a condition) to define variable parts of the URLs and then create the URLs using those variables.

Still I wouldn’t like the code that much. :wink:

```js
let urlPartOne
let urlPartTwo
let urlPartThree
let urlPartFour

if (result[day].rank == 'SOLEMNITY' || result[day].rank == 'FEAST') {
  urlPartOne = result[day].rank
  urlPartTwo = result[day].cycles.sundayCycle
  urlPartThree = result[day].key
  urlPartFour = '&nbsp;&nbsp;&nbsp;&nbsp;'  // Consider removing this
} else if (result[day].rank == 'SUNDAY') {
  urlPartOne = result[day].seasons
  urlPartTwo = result[day].rank
  urlPartThree = result[day].cycles.sundayCycle + '/' + String(result[day].calendar.weekOfSeason).padStart(2, '0') + '/' + result[day].calendar.dayOfWeek
  urlPartFour = '&nbsp;&nbsp;&nbsp;'  // Consider removing this

}

let calenderLink = url + urlPartOne + '/' + urlPartTwo + '/' + urlPartThree

// Note that I replaced the double qoutes (`"`) with single quotes (`'`) in order to removed the backslashes (`\`) from strings
mydescription += "<a href='" + calenderLink + ".mp3' target='_blank'><span style='font-size:12px;'>ஒலி</span></a><br>"
mydescription += "<a href='" + calenderLink + ".mp4' target='_blank'><span style='font-size:12px;'>ஒளி</span></a>" + urlPartFour
mydescription += "<a href='" + calenderLink + "R.mp3' target='_blank'><span style='font-size:12px;'>சிந்தனை</span><br></a>"

if (result[day].calendar.dayOfWeek == 0) {
  mydescription += "<a href='" + calenderLink + "M.pdf' target='_blank'><span style='font-size:12px;'>திருப்பலி</span><br></a>"
}
```

> Is there something in javascript that can take this three lines and put them in a variable that can be used in other place in the code?

It depends where exactly is the _other place_. If it is within the same file and you declare a function using `var` (as you already do), you declare it as global variable, i.e. it is accessible anywhere in that file.

**Reply by pvanthony** [reply]
@tukusejssirs This is a great idea. I did not think of this. Thank you for sharing.
@emagnier Thank you for the link. It is exactly what I was looking for.
Thank you both for helping. It has been really helpful.
